dali/pipeline/executor/executor.h

```diff
@@ -681,12 +698,12 @@ void Executor<WorkspacePolicy, QueuePolicy>::ShareOutputs(DeviceWorkspace *ws) {
     auto storage_dev = out_tensor.producer.storage_device;
     VALUE_SWITCH(storage_dev, storage_dev_static, (StorageDevice::GPU, StorageDevice::CPU),
     (
-      VALUE_SWITCH(op_type, op_type_static, (OpType::MIXED, OpType::GPU),
+      VALUE_SWITCH(op_type, op_type_static, (OpType::CPU, OpType::MIXED, OpType::GPU),
       (
         auto &queue = get_queue<op_type_static, storage_dev_static>(
             tensor_to_store_queue_[out_tensor_id]);
         auto stage_output_idx = output_idx[op_type_static];
-        ws->AddOutput(queue[stage_output_idx]);
+        ws->AddOutput(PresentAsTensorList(queue[stage_output_idx]));
```

Comment:
Did you try that when CPU queue != GPU queue size?

Reply:
I can try....

Reply:
Well.. this line doesn't change anything in the flow. Now the `queue[stage_output_idx])` could be TensorList or TensorVector, previously it was only TensorVector. So still the same ops are in the pipeline, but assigned to different devices and with different output type.